pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3247,42 +3247,44 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
 
     // Check that all the segments from 'segmentsFrom' exist in the table
     Set<String> segmentsForTable = new HashSet<>(getSegmentsFor(tableNameWithType, true));
-    Set<String> unavailableSegmentsInFrom = Sets.difference(new HashSet<>(segmentsFrom), segmentsForTable);
-    Preconditions.checkArgument(unavailableSegmentsInFrom.isEmpty(), String.format(
-        "'%s' from 'segmentsFrom' are unavailable in the table. (tableName = '%s', segmentsFrom = '%s', "
-            + "segmentsTo = '%s')", unavailableSegmentsInFrom, tableNameWithType, segmentsFrom, segmentsTo));
-
-    // Check that all the segments from 'segmentTo' does not exist in the table.
-    Set<String> availableSegmentsInTo = Sets.intersection(new HashSet<>(segmentsTo), segmentsForTable);
-    Preconditions.checkArgument(availableSegmentsInTo.isEmpty(), String.format(
-        "'%s' from 'segmentsTo' should not be available in the table at this point. (tableName = '%s', "
-            + "segmentsFrom = '%s', segmentsTo = '%s')", availableSegmentsInTo, tableNameWithType, segmentsFrom,
-        segmentsTo));
+    for (String segment : segmentsFrom) {
+      Preconditions.checkState(segmentsForTable.contains(segment),
```

Comment:
Do you think that we will ever need the full `segmentsFrom/segmentsTo` list for the error cases? The new code will never log the full `segmentsFrom/segmentsTo` in case of error.

Reply:
👍